prov/efa/src/efa_hmem.c

```diff
@@ -311,28 +312,28 @@ int efa_hmem_info_update_all(struct efa_domain *efa_domain)
 
 	ret = 0;
 
-	err = efa_hmem_info_update_system(&hmem_info[FI_HMEM_SYSTEM]);
+	err = efa_hmem_info_init_system(&hmem_info[FI_HMEM_SYSTEM], efa_domain);
 	if (err) {
 		ret = err;
 		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the System hmem_info struct! err: %d\n",
 			 err);
 	}
 
-	err = efa_hmem_info_update_cuda(&hmem_info[FI_HMEM_CUDA], efa_domain);
+	err = efa_hmem_info_init_cuda(&hmem_info[FI_HMEM_CUDA], efa_domain);
 	if (err) {
 		ret = err;
 		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Cuda hmem_info struct! err: %d\n",
 			 err);
 	}
 
-	err = efa_hmem_info_update_neuron(&hmem_info[FI_HMEM_NEURON], efa_domain);
+	err = efa_hmem_info_init_neuron(&hmem_info[FI_HMEM_NEURON], efa_domain);
 	if (err) {
 		ret = err;
 		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Neuron hmem_info struct! err: %d\n",
 			 err);
 	}
 
-	err = efa_hmem_info_update_synapseai(&hmem_info[FI_HMEM_SYNAPSEAI]);
+	err = efa_hmem_info_init_synapseai(&hmem_info[FI_HMEM_SYNAPSEAI], efa_domain);
 	if (err) {
```

Comment:
Note: this is currently unreachable since `efa_hmem_info_init_synapseai()` only returns `0`.